docs/pages/07_service.adoc

```diff
@@ -0,0 +1,955 @@
+    field String email = salesPerson.email ;
+}
+
+export PersonExport(Person person) {
+    service void delete() {
+        // implement this body in Java
+    }
+}
+----
+
+The complete authorization mechanism can be defined by carefully provisioning services to actors and transfer objects. Always be aware of that if a transfer object has already been received by an external user, then they are authorized to call the services provided by the transfer object.
+
+=== Guard
+
+A guard is a boolean expression that must evaluate to true in order to continue the execution.
+
+Guards can be defined for
+
+* services,
+* exports and
+* actors.
+
+To add a guard to services use `guard` as follows:
+
+*Example:*
+
+----
+export PersonExport(Person person) {
+    service void delete() guard person.leads!size() == 0;
+}
+----
+
+The example above defines a guard, and the delete service is not executed if that person has a lead.
+
+// TODO: what is the error message?
+
+To add a guard to exports use `guard` as follows:
+
+*Example:*
+
+----
+export PersonExport(Person person) guard person.leads!size() == 0 {
+    service void delete() ;
+    service Lead[] addLead(Lead lead) ;
+}
+----
+
+The example above defines a guard for export, and the services are not executed if that person has a lead.
+
+To add a guard to actors use `guard` as follows:
+
+*Example:*
+
+----
+actor SalesPersonActor(SalesPerson salesPerson)
+    realm "COMPANY"
+    identity salesPerson.email
+    guard salesPerson.leads!size() > 0;
+----
+
+The example above defines a guard for export, and the salesperson can only log in if he has at least one lead.
+
+=== CRUD annotations
+
+Simple create, delete and update services are provided automatically. We can use `@Create`, `@Delete` and `@Update` annotations to specify these services.
+
+
+Use `@Create` annotation on an action service to define a new create service. The create service creates a new transfer object as defined by its return type. The new transfer object will not be persisted, but only transfer object is returned.
+
+*Example:*
+
+----
+@Create
+service InvoiceTransfer myCreate();
+----
+
+@Create annotation can be used only on services that
+
+* have a mapped transfer object return type
+* have no parameter
+* have no body
+
+
+Use `@Delete` annotation on an action service to define a new delete service. The delete service deletes the entity referenced by its export's mapping field.
+
+*Example:*
+
+----
+@Delete
+service void myDelete();
+----
+
+@Delete annotation can be used only on services that
+
+* have their embedding export is mapped
+* have no return type
+* have no parameter
+* have no body
+
+
+Use `@Update` annotation on an action service to define a new update service. The update service persists the mapped transfer object passed as its argument. If the mapped entity of the transfer object does not exist, it creates a new entity. If the mapped entity already existed, it updates the entity.
+
```

Comment:
If the Update annotation also creates, I think the Create annotation is not necessary.